test/ModuleInterface/lifetime_dependence_test.swift

```diff
@@ -0,0 +1,34 @@
+// RUN: %empty-directory(%t)
+
+// RUN: %target-swift-frontend -swift-version 5 -enable-library-evolution -emit-module \
+// RUN:     -disable-experimental-parser-round-trip \
+// RUN:     -enable-experimental-feature NoncopyableGenerics \
+// RUN:     -enable-experimental-feature NonescapableTypes \
+// RUN:     -o %t/lifetime_dependence.swiftmodule \
+// RUN:     -emit-module-interface-path %t/lifetime_dependence.swiftinterface \
+// RUN:     %S/Inputs/lifetime_dependence.swift
+
+// Check the interfaces
+
+// RUN: %FileCheck %s < %t/lifetime_dependence.swiftinterface
+
+// See if we can compile a module through just the interface and typecheck using it.
+
+// RUN: %target-swift-frontend -compile-module-from-interface \
+// RUN:    -disable-experimental-parser-round-trip \
+// RUN:    -enable-experimental-feature NoncopyableGenerics \
+// RUN:    -enable-experimental-feature NonescapableTypes \
+// RUN:    %t/lifetime_dependence.swiftinterface -o %t/lifetime_dependence.swiftmodule
+
+// RUN: %target-swift-frontend -typecheck -I %t %s \
+// RUN:    -disable-experimental-parser-round-trip \
+// RUN:    -enable-experimental-feature NoncopyableGenerics \
+// RUN:    -enable-experimental-feature NonescapableTypes
+
+import lifetime_dependence
+
+// CHECK: @inlinable internal init(_ ptr: Swift.UnsafeRawBufferPointer, _ a: borrowing Swift.Array<Swift.Int>) -> _borrow(a) Self {
```

Comment:
If there are plans to adopt lifetime annotations in the stdlib (or any module in an SDK), the declarations using the annotations need to be guarded so that older compilers don't try to parse them.

Reply:
I think @kavon has already handled this. These are printed with:
`#if compiler(>=5.3) && $NoncopyableGenerics`

Reply:
Ok good! It would be helpful for the `CHECK:` lines in the test that the guards are in fact there, though.

Reply:
Done